### PR TITLE
docs: add warning about athena usage with exporter

### PR DESCRIPTION
**Pull request by tobiaszheller**
This PR adds Preview indication to Athena audit log and warning about use of it with events exporter.

### CODE REVIEW

File: docs/pages/reference/backends.mdx [hunk 1]
```diff
@@ -946,7 +946,32 @@ Config option priority is applied in the following order:
   are not supported in certain regions or environments or are only supported in GovCloud.
 </Admonition>
 
-## Athena
+## Athena (Preview)
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:949]: Are we still marking features as in preview? CC @r0mant

File: docs/pages/reference/backends.mdx [hunk 2]
```diff
@@ -946,7 +946,32 @@ Config option priority is applied in the following order:
...
+  title="Version warning"
+  opened={true}
+  min="14.0"
+>
+  Athena audit back-end is available starting from Teleport `14.0`.
+</Details>
+
+<Admonition type="warning" title="Preview">
+  Athena audit back-end is currently in Preview.
+</Admonition>
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:961]:
```suggestion
The Athena audit log back-end is available starting from Teleport v14.0, and is currently in preview.
```

There are a lot of boxes in a short span of content, which becomes pretty distracting. I've suggested condensing this content into regular body text.

File: docs/pages/reference/backends.mdx [hunk 3]
```diff
@@ -946,7 +946,32 @@ Config option priority is applied in the following order:
...
+
+<Notice type="danger">
+
+If you are using
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:965]: Users may not be familiar with the `SearchEvents` API. I would use more general language, for example, referring to Teleport integrations that rely on audit events.

Does this apply to access request plugins?

File: docs/pages/reference/backends.mdx [hunk 4]
```diff
@@ -946,7 +946,32 @@ Config option priority is applied in the following order:
...
+[Teleport-event-exporter](https://goteleport.com/docs/management/export-audit-events/fluentd/)
+or any other mechanism for polling of SearchEvents API, you should not use
+Athena audit back-end, because in Preview it can generate big costs.
+
+New version of Teleport Event Exporter is under development to address that
+issue.
```
Comment (ptgott) [inline docs/pages/reference/backends.mdx:971]:
```suggestion
the [Teleport event handler](https://goteleport.com/docs/management/export-audit-events/fluentd/), you should not use
the Athena audit back-end, as the preview release of this back-end is especially resource intensive when polling the Teleport Auth Service for audit events.

A new version of the Teleport event handler is under development to address this
issue.
```

**Review by ptgott — APPROVED** [review]

File: docs/pages/reference/backends.mdx (hunk 3 above)
Comment (tobiaszheller) [reply docs/pages/reference/backends.mdx]: > Does this apply to access request plugins?

No.

> Users may not be familiar with the SearchEvents API.

I have added link to Search Events API.

**Comment by tobiaszheller** [comment]
@klizhentas could you take a look at this changes?

**Review by klizhentas — APPROVED** [review]

File: docs/pages/reference/backends.mdx (hunk 1 above)
Comment (klizhentas) [reply docs/pages/reference/backends.mdx:949]: For this specific one, it's OK, because we know it has a serious implementation limitation.

**Pull request merged**